lib/config-base.d.ts

```diff
@@ -62,7 +62,7 @@ export interface HTTPOptions {
      */
     connectTimeout?: number;
     /**
-     * The number of milliseconds to wait before giving up on a connection attempt.
+     * The number of milliseconds to wait before giving up on a requested response from a service.
```

Comment:
Use the [XMLHttpRequest.timeout terminology from MDN](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/timeout).
```suggestion
     * The number of milliseconds a request can take before automatically being terminated.
```

Code which shows where httpOptions.timeout is used:
* Set in XMLHttpRequest.timeout in: https://github.com/aws/aws-sdk-js/blob/c62b6ed76c2f9676247e165bd3ed66daebec8f1c/lib/http/xhr.js#L70-L72
* Set in request.setTimeout in: https://github.com/aws/aws-sdk-js/blob/c62b6ed76c2f9676247e165bd3ed66daebec8f1c/lib/http/node.js#L82

Node.js docs:
* [request.setTimeout(timeout[, callback])](https://nodejs.org/api/http.html#http_request_settimeout_timeout_callback)
* [socket.setTimeout(timeout[, callback])](https://nodejs.org/api/net.html#net_socket_settimeout_timeout_callback)